### ISSUE TITLE
[Bug] InfoWindow renders twice after content load

### ISSUE THREAD

**Issue by Taltalonix** [visgl/react-google-maps#109]
### Description

When using the InfoWindow component it renders twice, once empty and then when the children components are rendered it refreshes.  

https://github.com/visgl/react-google-maps/assets/35221237/4b31b633-37c1-4fa5-aa18-fa635f9f1ab8

Wasn't sure if it's relevant enough to create a pull request, but this fixed the problem:
```tsx
/**
 * Props for the Info Window Component
 */
export type InfoWindowProps = google.maps.InfoWindowOptions & {
    onCloseClick?: () => void;
    anchor?: google.maps.Marker | google.maps.marker.AdvancedMarkerElement | null;
};

/**
 * Component to render a Google Maps Info Window
 */
export const InfoWindow = (props: PropsWithChildren<IMyInfoWindowProps>) => {
    const { children, anchor, onCloseClick, ...infoWindowOptions } = props;
    const map = useContext(GoogleMapsContext)?.map;

    const infoWindowRef = useRef<google.maps.InfoWindow | null>();

    const [contentContainer, setContentContainer] =
        useState<HTMLDivElement | null>(null);

    // create infowindow once map is available
    useEffect(() => {
        if (!map) return;

        if (!infoWindowRef.current) {
            infoWindowRef.current = new google.maps.InfoWindow(infoWindowOptions);
        }

        // Add content to info window
        const el = document.createElement('div');
        infoWindowRef.current.setContent(el);

        if (onCloseClick) {
            infoWindowRef.current.addListener('closeclick', () => {
                onCloseClick();
            });
        }

        setContentContainer(el);

        // Cleanup info window and event listeners on unmount
        return () => {
            if (infoWindowRef.current) {
                google.maps.event.clearInstanceListeners(infoWindowRef.current);
                infoWindowRef.current.close();
            }
            infoWindowRef.current = null;
            el.remove();

            setContentContainer(null);
        };
    }, [map, children, anchor]);

    // Open info window after content container is set
    useEffect(() => {
        if (contentContainer && infoWindowRef.current && anchor) {
            infoWindowRef.current.open({ map, anchor });
        }
    }, [contentContainer, infoWindowRef, anchor, map]);

    return (
        <>{contentContainer !== null && createPortal(children, contentContainer)}</>
    );
};
```

https://github.com/visgl/react-google-maps/assets/35221237/c203c932-bb5f-4525-bb57-e65ffef194db

### Steps to Reproduce

Just rendering the InfoWindow caused this issue, I will note that I used conditional rendering so the mounting functionality might be related to the issue:
```tsx
<Map {...mapProps}>
    {anchor && <InfoWindow anchor={anchor} />
</Map>
```

### Environment

- Library version: @vis.gl/react-google-maps@0.3.3
- Google maps version: weekly
- Browser and Version: Google Chrome Version 119.0.6045.160 (Official Build) (64-bit)
- OS: Windows 10

**Comment by usefulthink** [comment]
Do you have StrictMode enabled in your application? This sounds like it could have something to do with react running all effects twice in dev-mode..

**Comment by Taltalonix** [comment]
> Do you have StrictMode enabled in your application? This sounds like it could have something to do with react running all effects twice in dev-mode..

It is, but shouldn't it work even with StrictMode enabled?

**Comment by usefulthink** [comment]
You're probably right, the problem isn't so much that it gets rendered twice but that the infowindow is opened before the content has been rendered. Would you mind putting together a PR with your suggested fix?

**Comment by usefulthink** [comment]
This should be fixed as of v0.5.1
